permissions/src/main/java/com/google/accompanist/permissions/MutablePermissionState.kt

```diff
@@ -86,26 +86,41 @@ internal class MutablePermissionState(
     private val activity: Activity
 ) : PermissionState {
 
-    override var status: PermissionStatus by mutableStateOf(getPermissionStatus())
+    override var status: PermissionStatus by mutableStateOf(getPermissionStatus(false))
 
     override fun launchPermissionRequest() {
         launcher?.launch(
             permission
         ) ?: throw IllegalStateException("ActivityResultLauncher cannot be null")
     }
 
+    override fun openAppSettings() {
+        activity.openAppSettings(permission)
+    }
+
     internal var launcher: ActivityResultLauncher<String>? = null
 
     internal fun refreshPermissionStatus() {
-        status = getPermissionStatus()
+        status = getPermissionStatus(isPermissionPostRequest = true)
```

Comment:
`refreshPermissionStatus()` is also called from `PermissionLifecycleCheckerEffect`, so right now the initial lifecycle event refreshing the status when the permission isn't granted will result in a status of `PermanentlyDenied`.

Reply:
I can track `isPermissionPostRequest` as its own property and set it separately from `refreshPermissionStatus`, since once that is set to `true` there shouldn't be any scenario that would make it go back to `false`.

Reply:
Is there any ordering guarantees between `rememberLauncherForActivityResult` and `PermissionLifecycleCheckerEffect`?